newIDE/app/src/EventsFunctionsList/index.js

```diff
@@ -917,7 +954,17 @@ const EventsFunctionsList = React.forwardRef<
               {
                 icon: <Add />,
                 label: i18n._(t`Add a function`),
-                click: () => addNewEventsFunction(null),
+                click: () => {
+                  const index =
+                    !selectedEventsBasedBehavior &&
+                    !selectedEventsBasedObject &&
+                    selectedEventsFunction
+                      ? eventsFunctionsExtension.getEventsFunctionPosition(
+                          selectedEventsFunction
+                        ) + 1
+                      : eventsFunctionsExtension.getEventsFunctionsCount();
+                  addNewEventsFunction(null, null, null, index);
```

Comment:
At some point consider maybe using an object so that optional fields are less cryptic:
```suggestion
                  addNewEventsFunction({ index });
```